docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,172 @@
+
+
+There are 2 flows for completing installation.  One is managed by controller, and one by assisted service.
+
+In the flow managed by the assisted service, the following steps must be completed:
+
+- kubeconfig must be uploaded
+- cluster operators are successful
+- Monitored operators are either successful or failed
+
+In the flow managed by the controller all steps must be completed (in the table above).  The last step is a 
+a notification to complete installation.
+
+### Suggested changes
+- Only assisted service should notify installation completion
+- All steps by controller will be known as stages in the assisted service.  This will enable the service to manage 
+them in a similar way that host stages are managed.
+- Controller will not stop activity due to timeout.
+- Assisted service should be able to terminate the controller.
+- All existing timeouts except installation timeout and rebooting timeout should be treated as soft timeouts (i.e notification only)
+- In case of some optional operation (i.e managed operators), the user will be able to decide to skip it and set the cluster as completed.
+
+### Open Questions
+
+- Do we also soften the 24h `INSTALLATION_TIMEOUT` timeout?
+
+- Should we collect must gather logs when the installation timeout expires, even
+  if we don't mark it as failed?
+- Should any of these changes be optional ?
+
+### UI Impact
+
+The UI will need to explicitly show to the user that the cluster installation
+is taking longer than expected, and give suggestions on how to proceed. For
+example, we could present a warning message with this text:
+
+> Cluster installation is taking too long
+>
+> It should have completed in approximately 45 minutes, but it took 23 hours
+> and 52 minutes already. Check the logs to find out why or reset the cluster
+> to start over.
+
+The progress bars and other UI elements used to indicate progress should also
+explicitly indicate that the installation is taking longer than expected, for
+example using warning icons or specific colors. For example:
+
+![UI example](./soft-install-timeout/ui-example.png)
+
+### Test Plan
+
+We will need the following test cases, for the preparation, installation and
+finalizing phases:
+
+- Prepare a cluster that exceeds the timeout and verify that fixing the issue
+  manually allows the service to continue with the installation.
+
+- Verify that the UI explicitly shows the information about the expired
+  timeout, and that it recovers when the issue is eventually fixed and the
+  service continues with the installation.
+
+These tests may require introducing a mechanism to artificially delay the
+installation in the agent or the installer.
+
+## Drawbacks
+
+Most of our cluster installation failures are due to these timeouts. If we just
+disable them then we will have a large amount of installations that have failed
```

Comment:
Do we know for sure that the ones we mark today as failed are unrecoverably failed?

After implementing this proposal, if a user does nothing with the API or UI, it'll eventually be marked as failed anyway, right?

After implementing this proposal, if a user resets their installation, it seems that should still count as some sort of failure. If people start an installation and then have to abandon it and reset, that seems like some sort of failure worth tracking.

Reply:
We already have a fair amount of cancelations today that we can't explain.  This would add more "real" failures into the mix of other reasons why somebody would cancel.